File: js/leetcode/lc0009.js
var isPalindrome = function (x) {
    if (x < 0) return false;
    let y = 0;
    let v = x;
    while (v > 0) {
        y *= 10;
        y += v % 10;
        v = Math.floor(v / 10);
    }
    return x == y;
};

module.exports = {
    isPalindrome: isPalindrome
}

File: go/src/leetcode/lc0009.go
package leetcode

func isPalindrome(x int) bool {
	if x < 0 {
		return false
	}
	y := 0
	t := x
	for t > 0 {
		y *= 10
		y += t % 10
		t /= 10
	}
	return x == y
}
